### PR TITLE
Force resolve picker on blur or delayed response

### PR DESCRIPTION
**Pull request by martellaj**
#### Description of changes

We've received a few incidents in FR and internally that the people picker in the new sharing UI forces users to manually select users. This change force resolves the people picker under 2 circumstances:

- If an API call is made to fetch suggestions and user leaves the picker before the results return, the picker will auto-resolve to the first suggestion (if there is one).
- If suggestions have returned, but user leaves the picker (i.e. clicks away), the picker will auto-resolve to the first suggestion (if there is one).

### CODE REVIEW

**Comment by martellaj** [comment]
+ @joschect, this is the first PR for the changes we chatted about last week.

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx [hunk 1]
```diff
@@ -284,6 +285,16 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
       suggestedDisplayValue: itemValue,
       suggestionsVisible: this.input.value !== '' && this.input.inputElement === document.activeElement
     });
+
+    /**
+     * If user exits the input box before suggestions are returned,
+     * select the first result upon promise resolution, if a suggestion
+     * is available.
+     */
+    if (this.suggestionStore.hasSelectedSuggestion() &&
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: I think that this should only be if I press tab or enter, or something like that. I might click away, and click back and expect to be able to still edit.

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx [hunk 2]
```diff
@@ -292,6 +303,17 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
...
   }
 
+  /**
+   * Select the first suggestion if one is available when user leaves
+   * the input area.
+   */
+  @autobind
+  protected onBlur() {
+    if (this.suggestionStore.hasSelectedSuggestion()) {
+      this.addItemByIndex(0);
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: Same as above.

File: packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx (hunk 1 above)
Comment (martellaj) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: Would you be open to making this is a behavior that you can opt in to? This behavior is exactly how our people picker has worked in the past and how customers are expecting it to work now and why we are receiving escalations and negative feedback.

Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: I think it should be an option then, maybe. Part of me feels that the picker should work more like a search box.

Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: But that it should also work the way you mention.

Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: Actually, you're right. Ignore me.

Comment (martellaj) [inline packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx]: Thanks for the review, Jon!

**Pull request merged**